test/tasks/maintenance/backfill_linkset_links_to_version_metadata_task_test.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+require "test_helper"
+
+class Maintenance::BackfillLinksetLinksToVersionMetadataTaskTest < ActiveSupport::TestCase
+  context "#collection" do
+    should "return all versions" do
+      assert_equal Version.count, Maintenance::BackfillLinksetLinksToVersionMetadataTask.collection.count
+    end
+  end
+
+  context "#process" do
+    context "without a linkset" do
+      setup do
+        @version = create(:version)
+        @rubygem = @version.rubygem
+        @rubygem.update!(linkset: nil)
+      end
+
+      should "not change version metadata" do
+        assert_no_changes "@version.reload.metadata" do
+          Maintenance::BackfillLinksetLinksToVersionMetadataTask.process(@version)
+        end
+      end
+    end
+
+    context "with a linkset and version metadata uris" do
+      setup do
+        @version = create(
+          :version,
+          metadata: {
+            "source_code_uri" => "https://example.com/source",
+            "documentation_uri" => "https://example.com/docs",
+            "foo" => "bar"
+          }
+        )
+        @rubygem = @version.rubygem
+        @rubygem.linkset.update!("home" => "https://example.com/home",
+                                 "wiki" => "https://example.com/wiki")
+      end
+
+      should "only update the home uri" do
```

Comment:
Why only home?

Reply:
Because in the Links model we only use the home uri from the link set when version metadata has uris defined, and ignore all the other links in the link set

Reply:
So the home link is the only one that currently falls back, and if there are any URIs already in metadata, they are being ignored right now, so we don't bother to copy them. Then after this we remove that fallback and remove linkset entirely?

Makes sense, thanks for explaining.

Reply:
yeah, it's confusing. which is why i think it would be nice to get rid of link set entirely, because of the "sometimes but not always used" situation